contracts/token/non-fungible/src/non_fungible.rs

```diff
@@ -0,0 +1,343 @@
+    ///   exist.
+    ///
+    /// # Events
+    ///
+    /// * topics - `["transfer", from: Address, to: Address]`
+    /// * data - `[token_id: u128]`
+    ///
+    /// # Notes
+    ///
+    /// We recommend using [`crate::transfer()`] when implementing this
+    /// function.
+    fn transfer(e: &Env, from: Address, to: Address, token_id: u128);
+
+    /// Transfers `token_id` token from `from` to `to` with authorization from
+    /// `spender`.
+    ///
+    /// Unlike `transfer()`, which is used when the token owner initiates the
+    /// transfer, `transfer_from()` allows an approved third party
+    /// (`spender`) to transfer the token on behalf of the owner. This
+    /// function includes an on-chain check to verify that `spender` has the
+    /// necessary approval.
+    ///
+    /// WARNING: Note that the caller is responsible to confirm that the
+    /// recipient is capable of receiving the `Non-Fungible` or else the NFT
+    /// may be permanently lost.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `spender` - The address authorizing the transfer.
+    /// * `from` - Account of the sender.
+    /// * `to` - Account of the recipient.
+    /// * `token_id` - Token id as a number.
+    ///
+    /// # Errors
+    ///
+    /// * [`NonFungibleTokenError::IncorrectOwner`] - If the previous owner is
+    ///   not `from`.
+    /// * [`NonFungibleTokenError::UnauthorizedTransfer`] - If the caller does
+    ///   not have the right to approve.
+    /// * [`NonFungibleTokenError::NonexistentToken`] - If the token does not
+    ///   exist.
+    ///
+    /// # Events
+    ///
+    /// * topics - `["transfer", from: Address, to: Address]`
+    /// * data - `[token_id: u128]`
+    ///
+    /// # Notes
+    ///
+    /// We recommend using [`crate::transfer_from()`] when implementing this
+    /// function.
+    fn transfer_from(e: &Env, spender: Address, from: Address, to: Address, token_id: u128);
+
+    /// Gives permission to `to` to transfer `token_id` token to another
+    /// account. The approval is cleared when the token is transferred.
+    ///
+    /// Only a single account can be approved at a time.
+    /// To remove an approval, the owner can approve their own address,
+    /// effectively removing the previous approved address.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to Soroban environment.
+    /// * `owner` - The address holding the tokens.
+    /// * `to` - Account of the recipient.
+    /// * `token_id` - Token id as a number.
+    /// * `live_until_ledger` - The ledger number at which the allowance
+    ///   expires.
+    ///
+    /// # Errors
+    ///
+    /// * [`NonFungibleTokenError::NonexistentToken`] - If the token does not
+    ///   exist.
+    /// * [`NonFungibleTokenError::InvalidApprover`] - If `auth` (param of
+    ///   [`Erc721::_approve`]) does not have a right to approve this token.
```

Comment:
no need to ref erc721

Reply:
addressed in [7552cf7](https://github.com/OpenZeppelin/stellar-contracts/pull/75/commits/7552cf74f0cf98b1838cc4af1e01e3451fd69904)